helix-term/src/ui/document.rs

```diff
@@ -178,10 +181,16 @@ pub fn render_text<'t>(
 
     let (mut formatter, mut first_visible_char_idx) =
         DocumentFormatter::new_at_prev_checkpoint(text, text_fmt, text_annotations, offset.anchor);
-    let mut styles = StyleIter {
+    let mut syntax_styles = StyleIter {
         text_style: renderer.text_style,
         active_highlights: Vec::with_capacity(64),
-        highlight_iter,
+        highlight_iter: syntax_highlight_iter,
+        theme,
+    };
+    let mut overlay_styles = StyleIter {
+        text_style: renderer.text_style,
```

Comment:
using ``render.text_style`` is wrong here it will set the whole document to the text colour as the overlay is patched last, this should just be ``Style::default`` so if there is no style here we don't override the past text highlighting

Reply:
see StyleIter for why it will highlight all the text the default text color: https://github.com/helix-editor/helix/blob/41b307b673a34183123585d63746cb756c1779ed/helix-term/src/ui/document.rs#L64-L66 as overlay_highlight_iter covers the whole viewport

Reply:
Looking into this more deeply, this text_style is set based on the given `ui.text` scope. The documentation about this says "Command prompts, popup text, etc.", but as this is really the default text style that is used, we should probably change that description.